pipeline-runner/tests/testthat/test-gem2s-3-run_emptydrops.R

```diff
@@ -0,0 +1,35 @@
+mock_counts <- function() {
+    set.seed(0)
+    DropletUtils:::simCounts()
+}
+
+
+test_that("run_emptydrops is skipped when dataset is pre-filtered", {
+    counts <- mock_counts()
+    ntot <- Matrix::colSums(counts)
+    counts <- counts[, ntot > 100]
+
+    prev_out <- list(counts_list = list(sample1=counts))
+    expect_message(
+        out <- run_emptydrops(NULL, NULL, prev_out),
+        'filtered'
+    )
+
+    expect_equal(out$output$edrops, list())
+})
+
+test_that("run_emptydrops runs when not pre-filtered", {
+    counts <- mock_counts()
+    prev_out <- list(counts_list = list(sample1=counts, sample2=counts))
+    out <- run_emptydrops(NULL, NULL, prev_out)$output
```

Comment:
what about a test for running emptydrops when filter 2 or filter 1 or both have been skipped previously?

Reply:
GEM2S doesn't have the concept of skipping/disabling previous steps. All steps of GEM2s run every time.

Reply:
yes, of course, forgot that